src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CallbackCcdApi.java

```diff
@@ -0,0 +1,72 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import feign.FeignException;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
+
+import javax.annotation.Nonnull;
+
+import static java.lang.String.format;
+
+@Component
+public class CallbackCcdApi {
```

Comment:
Class not really deals/represents any callback reference whatsoever. Perhaps just rename as `CcdApi` and it'll stand as a wrapper. one and only place which interacts with ccd?

Actually. It's just mimicking the event publisher. Can just create custom event, define relevant event id (attachScannedDocs) and that's it. I'm happy to leave as is as long as this task works and this is marked as tech debt. Defo needs a proper review after everything is done

Reply:
yup ... thats what its evolving to be but at the moment its highly coupled with Callback.
So at the end of the evolution of the code we should merge in all the calls to the ccdApi and rename this to the CcdApi for the Adapter onto the external dependency.

I suppose I can rename it now ... but was planning on heading that direction later with the merging of the CaseRetriever and this class.

Reply:
Renaming also not relevant atm either. It's a continuous non-breaking development. Surely easier to investigate later and make proper refinement. Confirming this one now

Reply:
Done.